Support Firestore as LlamaIndex Storage

pl04351820 opened the pull request:
# Description

This PR adds support for using Google Firestore as LlamaIndex KV/Doc/Index Storage.

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Added test coverage for all new storage supported on top of the Firestore.

- [X] Added new unit/integration tests

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

logan-markewich commented:
This looks good! Just a few linting errors with `mypy` to clean up

Would it be possible to add a demo notebook in `docs/examples/docstore` and update the docs in `docs/core_modules/data_modules/storage/docstores.md` ?

pl04351820 commented:
- Fixed the lint the errors in firestore_kvstore.py;
- Added a demo notebook in docs/examples/docstore;
- Updated docs/core_modules/data_modules/storage/docstores.md to reflect the change.

pl04351820 commented:
Applied the lint fix for tests/storage/index_store/test_firestore_indexstore.py
